lib/client/workflow-client/workflowClient.js

```diff
@@ -0,0 +1,367 @@
+var q = require('q');
+var _ = require('lodash');
+var shortid = require('shortid');
+var CONSTANTS = require('../../constants');
+var MediatorTopicUtility = require('fh-wfm-mediator/lib/topics');
+
+/**
+ *
+ * Getting Promises for done and error topics.
+ * This will resolve or reject the returned promise depending on the topic published.
+ *
+ *
+ * TODO: This may be of more use in fh-wfm-mediator...
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {Promise}
+ */
+function getTopicPromises(doneTopicPromise, errorTopicPromise) {
+  var deferred = q.defer();
```

Comment:
This is going to move as part of https://issues.jboss.org/browse/RAINCATCH-629 . We can probably do it then. Don't want to switch everything to bluebird in this PR.